core/src/main/java/jenkins/management/AdministrativeMonitorsDecorator.java

```diff
@@ -0,0 +1,136 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.management;
+
+import hudson.Extension;
+import hudson.Functions;
+import hudson.diagnosis.ReverseProxySetupMonitor;
+import hudson.model.AdministrativeMonitor;
+import hudson.model.PageDecorator;
+import hudson.util.HudsonIsLoading;
+import hudson.util.HudsonIsRestarting;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.Ancestor;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Show a notification and popup for active administrative monitors on all pages.
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class AdministrativeMonitorsDecorator extends PageDecorator {
+    private final Collection<String> ignoredJenkinsRestOfUrls = new ArrayList<>();
+
+    public AdministrativeMonitorsDecorator() {
+        // redundant
+        ignoredJenkinsRestOfUrls.add("manage");
+
+        // otherwise this would be added to every internal context menu building request
+        ignoredJenkinsRestOfUrls.add("contextMenu");
+    }
+
+    @Override
+    public String getDisplayName() {
+        return Messages.AdministrativeMonitorsDecorator_DisplayName();
+    }
+
+    public int getActiveAdministrativeMonitorsCount() {
+        return getActiveAdministrativeMonitors().size();
+    }
+
+    public Collection<AdministrativeMonitor> getActiveAdministrativeMonitors() {
+        Collection<AdministrativeMonitor> active = new ArrayList<>();
+        Collection<AdministrativeMonitor> ams = new ArrayList<>(Jenkins.getInstance().administrativeMonitors);
+        for (AdministrativeMonitor am : ams) {
+            if (am instanceof ReverseProxySetupMonitor) {
+                // TODO make reverse proxy monitor work when shown on any URL
+                continue;
+            }
+            if (am.isEnabled() && am.isActivated()) {
```

Comment:
Nice2have: Catch runtime exceptions (just in case somebody provides unstable extension point implementation)